PANDORA/Pandora/Modelling_functions.py

```diff
@@ -224,134 +231,142 @@ def predict_anchors_netMHCpan(peptide, allele_type,
-                try:
-                    pred[ln[1]].append((ln[3], float(ln[12])))
-                except KeyError:
-                    pred[ln[1]] = [(ln[3], float(ln[12]))]
+    # Retrieves the enviroment of the netMHCpan script and the path of the directory where this script is in 
+    if os.getenv('netMHCpan', default=None):
+        netmhcpan_file_path = os.getenv('netMHCpan', default=None)
+        netmhcpan_path = os.path.dirname(netmhcpan_file_path)
 
-    # Sort each allele result per Rank_EL
-    for allele in pred:
-        pred[allele] = list(sorted(pred[allele], key=lambda x:x[1]))
-    
-    if len(pred) == 0:
-        print('ERROR: NetMHCpan-4.1 was not able to find any binding core for')
-        print('the provided peptide and MHC allele')
-        return None
-    
-    # For every allele, the binding core is predicted. Take the allele with the highest reliability score
-    best_allele = min((pred[i][0][1], i) for i in pred)[1]
-    
-    # Do a quick alignment of the predicted core and the peptide to find the anchors. (the predicted binding core can
-    # contain dashes -. Aligning them makes sure you take the right residue as anchor.
-    alignment = pairwise2.align.globalxx(peptide, pred[best_allele][0][0])
-    
-    #If there are multiple possible solutions, take the one with no gap at the anchor (second) position
-    if len(alignment)>1:
-        flag = False
-        #Search for the options without gap in the second postions
-        for prediction in alignment:
-            if prediction[1][1] != '-' and prediction[0][1] != '-':
-                pept1 = prediction[0]
-                pept2 = prediction[1]
-                flag = True
+        all_netMHCpan_alleles = []
+        with open(netmhcpan_path + '/data/allelenames') as f:
+            for line in f:
+                all_netMHCpan_alleles.append(line.split(' ')[0])#.replace(':',''))
+        
+        ## Format alleles
+        target_alleles = [i.replace('*','') for i in allele_type]
+        ## Make sure only netMHCpan available alleles are used
+        target_alleles = [i for i in target_alleles if i in all_netMHCpan_alleles]
+        
+        if len(target_alleles) == 0:
+            print('ERROR: The provided Target allele is not available in NetMHCpan-4.1')
+            return None
+        
+        target_alleles_str = ','.join(target_alleles)
+        
+        # Setup files
+        infile = netmhcpan_path + '/tmp/%s_%s_%s.txt' %(peptide, target_alleles[0].replace('*','').replace(':',''), datetime.today().strftime('%Y%m%d_%H%M%S'))
+        outfile = netmhcpan_path + '/tmp/%s_%s_%s_prediction.txt' %(peptide, target_alleles[0].replace(':',''), datetime.today().strftime('%Y%m%d_%H%M%S'))
+        
+        # Write peptide sequence to input file for netMHCIIpan
+        with open(infile, 'w') as f:
+            f.write(peptide)
+        
+        os.system('%s -p %s -a %s > %s' %(netmhcpan_file_path, infile, target_alleles_str, outfile))
```

Comment:
Please change all the "os.system(<command>)" in the functions you change to "subprocess.check_call(<command>, shell=True)"